cpp/src/column/column.cu

```diff
@@ -0,0 +1,197 @@
+      _data.data(), static_cast<bitmask_type const *>(_null_mask.data()),
+      _null_count,  0,
+      child_views};
+}
+
+// Create mutable view
+mutable_column_view column::mutable_view() {
+  // create views of children
+  std::vector<mutable_column_view> child_views;
+  child_views.reserve(_children.size());
+  for (auto const &c : _children) {
+    child_views.emplace_back(*c);
+  }
+
+  // Store the old null count
+  auto current_null_count = _null_count;
+
+  // The elements of a column could be changed through a `mutable_column_view`,
+  // therefore the existing `null_count` is no longer valid. Reset it to
+  // `UNKONWN_NULL_COUNT` forcing it to be recomputed on the next invocation of
+  // `null_count()`.
+  set_null_count(UNKNOWN_NULL_COUNT);
+
+  return mutable_column_view{_type,
+                             _size,
+                             _data.data(),
+                             static_cast<bitmask_type *>(_null_mask.data()),
+                             current_null_count,
+                             0,
+                             child_views};
+}
+
+// If the null count is known, return it. Else, compute and return it
+size_type column::null_count() const {
+  if (_null_count > UNKNOWN_NULL_COUNT) {
+    return _null_count;
+  } else {
+    // If the null mask isn't allocated, then we can return 0
+    if (0 == _null_mask.size()) {
+      _null_count = 0;
+      return null_count();
+    }
+    CUDF_FAIL(
+        "On-demand computation of null count is not currently supported.");
+  }
+}
+
+void column::set_null_count(size_type new_null_count) {
+  if (new_null_count > 0) {
+    CUDF_EXPECTS(nullable(), "Invalid null count.");
+  }
+  _null_count = new_null_count;
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief Copies the bits starting at the specified offset from a source
+ * bitmask into the destination bitmask.
+ *
+ * Bit `i` in `destination` will be equal to bit `i + offset` from `source`.
+ *
+ * @param destination The mask to copy into
+ * @param source The mask to copy from
+ * @param bit_offset The offset into `source` from which to begin the copy
+ * @param number_of_bits The number of bits to copy
+ *---------------------------------------------------------------------------**/
+__global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
+                                    bitmask_type const *__restrict__ source,
+                                    size_type bit_offset,
+                                    size_type number_of_bits) {
+  constexpr size_type warp_size{32};
+  size_type destination_bit_index = threadIdx.x + blockIdx.x * blockDim.x;
+
+  auto active_mask =
+      __ballot_sync(0xFFFF'FFFF, destination_bit_index < number_of_bits);
+
+  while (destination_bit_index < number_of_bits) {
+    bitmask_type const new_mask_element = __ballot_sync(
+        active_mask, bit_is_set(source, bit_offset + destination_bit_index));
```

Comment:
Wouldn't it be cheaper (in performance and code) to have only the first and last warps copy offset, and all other warps copy aligned, full 32-bit mask elements?  Shouldn't ever have to check whether an individual bit is set, and shouldn't need a thread per bit. Similar to `cudf::copy_range()`.

Reply:
Unless the `offset % 32 == 0`, then you'll either be reading from or writing to the interior of a word. I don't see how it's possible to copy entire words at a time.

`copy_range` does the same thing: https://github.com/rapidsai/cudf/blob/branch-0.10/cpp/src/copying/copy_range.cuh#L80

Reply:
Yeah, not exactly what I meant. I'll leave it for future optimization.